signature/types.go

```diff
@@ -197,17 +198,21 @@ func (signerInfo *SignerInfo) ExtendedAttribute(key string) (Attribute, error) {
 	return Attribute{}, errors.New("key not in ExtendedAttributes")
 }
 
-// AuthenticSigningTime returns the authentic signing time
+// AuthenticSigningTime returns the authentic signing time under signing scheme
+// notary.x509.signingAuthority.
+// For signing scheme notary.x509, since it only supports authentic timestamp,
+// an error is returned.
+//
+// Reference: https://github.com/notaryproject/specifications/blob/3b0743cd9bb99faee60600dc31d706149775fd49/specs/signature-specification.md#signing-time--authentic-signing-time
 func (signerInfo *SignerInfo) AuthenticSigningTime() (time.Time, error) {
-	switch signerInfo.SignedAttributes.SigningScheme {
+	switch signingScheme := signerInfo.SignedAttributes.SigningScheme; signingScheme {
 	case SigningSchemeX509SigningAuthority:
-		return signerInfo.SignedAttributes.SigningTime, nil
-	case SigningSchemeX509:
-		if len(signerInfo.UnsignedAttributes.TimestampSignature) > 0 {
-			// TODO: Add TSA support for AutheticSigningTime
-			// https://github.com/notaryproject/notation-core-go/issues/38
-			return time.Time{}, errors.New("TSA checking has not been implemented")
+		signingTime := signerInfo.SignedAttributes.SigningTime
+		if signingTime.IsZero() {
+			return time.Time{}, fmt.Errorf("authentic signing time must be present under signing scheme %q", signingScheme)
```

Comment:
Is it must under the spec or its optional? AFAIR the previous version of spec said that authentic signing time can be used only under `SigningSchemeX509SigningAuthority` but it was optional

Reply:
@priteshbandi Yes, it's a `REQUIRED` field according to the spec: https://github.com/notaryproject/specifications/blob/main/specs/signature-specification.md#standard-attributes

One thing to point out is that, we are still using the `SigningTime` field under `SigningSchemeX509SigningAuthority` as authentic signing time. However, based on spec, we should use the `Authentic Signing Time` field instead. (both fields are `REQUIRED`)